### ISSUE TITLE
Installation results in top-level "tests" package of sub-dependency 

### ISSUE THREAD

**Issue by KPrasch** [nucypher/nucypher#364]
A typical installation of coincurve (a sub-dependency of nucypher) can install it's own top-level test directory causing `import tests` to import the coincurve tests as a package.

It looks like this is because: 
  - There is a `__init__.py` in the coincurve test directory" https://github.com/ofek/coincurve/tree/master/tests
- `setup.py` does not exclude this package: https://github.com/ofek/coincurve/blob/master/setup.py#L280

Here are two ways to reproduce the issue:

##### Installing nucypher with pipenv

```
kieran@pegasus:~/Git/nucypher-kms$ pipenv install --dev --skip-lock 
...

kieran@pegasus:~/Git/nucypher-kms$ pipenv install --skip-lock -e . 
...

kieran@pegasus:~/Git/nucypher-kms$ pipenv run python 
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tests 
>>> tests.__file__
'/home/kieran/.local/share/virtualenvs/nucypher-kms-PHpd9bvk/lib/python3.6/site-packages/tests/__init__.py'
>>> quit()

kieran@pegasus:~/Git/nucypher-kms$ ls -la /home/kieran/.local/share/virtualenvs/nucypher-kms-PHpd9bvk/lib/python3.6/site-packages/tests
total 40
drwxr-xr-x   3 kieran kieran  4096 Jul 15 12:30 .
drwxr-xr-x 250 kieran kieran 12288 Jul 15 12:45 ..
-rw-r--r--   1 kieran kieran     0 Jul 15 12:30 __init__.py
drwxr-xr-x   2 kieran kieran  4096 Jul 15 12:30 __pycache__
-rw-r--r--   1 kieran kieran  3002 Jul 15 12:30 samples.py
-rw-r--r--   1 kieran kieran   166 Jul 15 12:30 test_ecdsa.py
-rw-r--r--   1 kieran kieran  4792 Jul 15 12:30 test_keys.py
-rw-r--r--   1 kieran kieran  2464 Jul 15 12:30 test_utils.py

```

##### Isolated Coincurve Installation
```
kieran@pegasus:~/Git/brokenCoinCurve$ pipenv install coincurve 
Creating a virtualenv for this project...
...

kieran@pegasus:~/Git/brokenCoinCurve$ pipenv run python
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tests
>>> tests.__file__
'/home/kieran/.local/share/virtualenvs/brokenCoinCurve-mYIv09Xr/lib/python3.6/site-packages/tests/__init__.py'
>>> quit()

kieran@pegasus:~/Git/brokenCoinCurve$ ls -la /home/kieran/.local/share/virtualenvs/brokenCoinCurve-mYIv09Xr/lib/python3.6/site-packages/tests
total 32
drwxr-xr-x  3 kieran kieran 4096 Jul 15 12:38 .
drwxr-xr-x 20 kieran kieran 4096 Jul 15 12:38 ..
-rw-r--r--  1 kieran kieran    0 Jul 15 12:38 __init__.py
drwxr-xr-x  2 kieran kieran 4096 Jul 15 12:38 __pycache__
-rw-r--r--  1 kieran kieran 3002 Jul 15 12:38 samples.py
-rw-r--r--  1 kieran kieran  166 Jul 15 12:38 test_ecdsa.py
-rw-r--r--  1 kieran kieran 4792 Jul 15 12:38 test_keys.py
-rw-r--r--  1 kieran kieran 2464 Jul 15 12:38 test_utils.py

```

Any help on this is appreciated.

**Comment by KPrasch** [comment]
Placing a `__init__.py` in our tests directory seems to override any other packages named "tests".

**Comment by KPrasch** [comment]
I opened an Issue over there, waiting for feedback

**Comment by KPrasch** [comment]
Fixed in coincurve #23

**Comment by AndreMiras** [comment]
Thanks @KPrasch, your link is pointing to nucypher issue which is a bit confusing, here's the coincurve link https://github.com/ofek/coincurve/pull/23
